### PR TITLE
PWN-4704 PWN-4705 PWN-4709 PWN-4711 - Onboarding error handling fixes

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-4704
https://p2pvalidator.atlassian.net/browse/PWN-4705
https://p2pvalidator.atlassian.net/browse/PWN-4709
https://p2pvalidator.atlassian.net/browse/PWN-4711

## Description of Work

The text in the toast is “Something went wrong. Please try again“ is not fully displayed
Infinite processing is displayed in the button after receiving an error on the request
The toast with an error is not displayed on the “A wallet found“ screen if an error has been received
An endless processing displays on the “A wallet found“ screen via reauthenticating

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/auth/gateway/parser/CountryCodeHelper.kt [hunk 1]
```diff
@@ -52,9 +53,11 @@ class CountryCodeHelper(
 
     private fun getMaskForCountryCode(countryCode: String, phoneCode: String): String {
         return try {
-            val exampleNumber = phoneNumberUtil.getExampleNumber(countryCode)
-            val internationalFormat =
-                phoneNumberUtil.format(exampleNumber, PhoneNumberUtil.PhoneNumberFormat.INTERNATIONAL)
+            val exampleNumber: Phonenumber.PhoneNumber? = phoneNumberUtil.getExampleNumber(countryCode)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/gateway/parser/CountryCodeHelper.kt:56]: found NPE exception in Sentry and decided to put explicit type here

File: app/src/main/java/org/p2p/wallet/auth/ui/restore/found/WalletFoundFragment.kt [hunk 2]
```diff
@@ -67,19 +67,17 @@ class WalletFoundFragment :
...
+    override fun onSuccessfulSignUp() {
         view?.post {
-            setLoadingState(isScreenLoading = false)
-            showErrorSnackBar(error)
+            replaceFragment(PhoneNumberEnterFragment.create())
         }
     }
 
-    override fun onSuccessfulSignUp() {
-        setLoadingState(isScreenLoading = false)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/restore/found/WalletFoundFragment.kt:78]: loading states changes are called in the presenter now

File: app/src/main/java/org/p2p/wallet/auth/ui/restore/found/WalletFoundFragment.kt [hunk 3]
```diff
@@ -90,22 +88,19 @@ class WalletFoundFragment :
     }
 
     private fun handleSignResult(result: ActivityResult) {
-        signInHelper.handler = this
-        try {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/restore/found/WalletFoundFragment.kt:94]: no need to try {} here

File: app/src/main/java/org/p2p/wallet/auth/web3authsdk/GoogleSignInHelper.kt [hunk 4]
```diff
@@ -55,18 +50,23 @@ class GoogleSignInHelper(
...
             getSignInClient(context).getSignInCredentialFromIntent(result.data)
         } catch (ex: ApiException) {
+            Timber.e(ex, "Error on getting Credential from result")
+
             if (shouldErrorBeHandled(ex)) {
                 if (connectionStateProvider.hasConnection()) {
-                    handler?.onCommonError(ex.message ?: ex.toString())
+                    errorHandler.onCommonError()
                 } else {
-                    handler?.onConnectionError(resourcesProvider.getString(R.string.onboarding_offline_error))
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/web3authsdk/GoogleSignInHelper.kt:66]: it's not for the Helper to decide what error to show, i moved it to a view

**Review by kabramovp2p — APPROVED** [review]

**Review by kuznetsov — APPROVED** [review]

**Pull request merged**